file_paths.md

```diff
@@ -163,156 +163,167 @@ and `pxp-agent`, all running as `LocalSystem` by default. It will not
         environments *                        # :environmentpath
           production *
             environment.conf *
+            hieradata *
             manifests *
             modules *
-        hiera.yaml *                          # :hiera_config
-        hieradata *                           # n/a
-        modules *                             # user modulepath
+        hiera.yaml                            # :hiera_config
+        modules                               # user modulepath
 
-    C:\ProgramData\PuppetLabs\facter *                                       same
+    C:\ProgramData\PuppetLabs\mcollective *
+        client.cfg *
+        facts.yaml *
+        server.cfg *
+
+    C:\ProgramData\PuppetLabs\puppet          # :confdir
+        auth.conf *                           # :rest_authconfig
+        autosign.conf                         # :autosign
+        binder_config.yaml                    # :binder_config
+        csr_attributes.yaml                   # :csr_attributes
+        custom_trusted_oid_mapping.yaml       # :trusted_oid_mapping_file
+        device.conf                           # :deviceconfig
+        fileserver.conf                       # :fileserverconfig
+        puppet.conf *                         # :config
+        routes.yaml                           # :route_file
+        ssl                                   # :ssldir
+
+    C:\ProgramData\PuppetLabs\pxp-agent
+        modules *                             # stores configuration files for pxp-agent modules
+            pxp-module-puppet.conf *          # configuration file of the pxp module puppet (optionally, to override puppet.bat location)
+        pxp-agent.conf                        # pxp-agent configuration file
+
+    C:\ProgramData\PuppetLabs\var
+        log *
+          mcollective.log
+          puppet.log
+          pxp-agent.log
+        run *
+          agent.pid
+          mcollective.pid
+          pxp-agent.pid
+
+    C:\Program Files\Puppet Labs\bin *        # symlink targets of puppet related binaries
+        environment.bat@ *                    -> C:\Program Files\Puppet labs\puppet\bin\environment.bat
+        puppet.bat@ *                         -> C:\Program Files\Puppet labs\puppet\bin\puppet.bat
+        hiera.bat@ *                          -> C:\Program Files\Puppet labs\puppet\bin\hiera.bat
+        mco.bat@ *                            -> C:\Program Files\Puppet labs\puppet\bin\mco.bat
+        facter.bat@ *                         -> C:\Program Files\Puppet labs\puppet\bin\facter.bat
+        puppet_shell.bat@ *                   -> C:\Program Files\Puppet labs\puppet\bin\puppet_shell.bat
+        run_facter_interactive.bat@ *         -> C:\Program Files\Puppet labs\puppet\bin\run_facter_interactive.bat
+        run_puppet_interactive.bat@ *         -> C:\Program Files\Puppet labs\puppet\bin\run_puppet_interactive.bat
+
+    C:\Program Files\Puppet Labs\facter *
```

Comment:
Typically you want to keep user data out of Program Files - that is what ProgramData is for. Let's be sure facts.d isn't something we would consider user data.

Basically the argument is that if files/directories are expected to be added/changed after install - they go to ProgramData. cc @jpogran @joshcooper